### ISSUE TITLE
Question: Is it bug in remove_cppwinrt_foundation_types()

### ISSUE THREAD

**Issue by jonitis** [Microsoft/xlang#225]
Hello,

library/impl/meta_reader/cache.h cache::remove_cppwinrt_foundation_types() has strange code that calls

```
  remove(members.classes, name);
```

twice. Seems to be copy-paste bug. Instead it should call

```
  remove(members.attributes, name);
  remove(members.contracts, name);
```

since attributes are subset of classes and contracts subset of structs. Should not really change anything with currently used blacklist, though.

**Comment by jonitis** [comment]
P.S.
I was playing with modification that does not store vector<TypeDef> in cache for interfaces, classes etc., but just reference_wrapper<TypeDef const>, since original TypeDef is always present in types map. But the win is actually negligible for 6.5 Mb windows.winmd:

```
namespaces:     338
TypeDefs:       13543 (216688)
  interfaces:   7394 (118304)
  classes:      4199 (67184)
  enums:        1578 (25248)
  structs:      110 (1760)
  delegates:    134 (2144)
  attributes:   44 (704)
  contracts:    84 (1344)
Memory saving:  216688 - 108344 = 108344
```

On x64 it is 16 bytes for row_base<TypeDef> vs. 8 bytes for TypeDef pointer. IMHO totally not worth the change. Would like to know your opinion too.

**Comment by kennykerr** [comment]
Thanks for reporting. Yes, that's a bug and I'll get it cleaned up. It's harmless though, which is why we didn't pick it up already.

**Comment by oldnewthing** [comment]
The duplicate is fixed by #238
